internal/app/kube_helpers.go

```diff
@@ -134,6 +135,64 @@ spec:
 
 }
 
+func setQuotas(ns string, quotas *quotas) {
+	if quotas == nil {
+		return
+	}
+
+	definition := `
+---
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: resource-quota
+spec:
+  hard:
+`
+	if quotas.Pods != "" {
```

Comment:
I just copy pasted what I had from the old PR before helm 3, will change

Reply:
Looking at it, the marshalling won't work because it isn't the exact same object format because of the customResources option

Reply:
what matters is the format we use in the Helmsman file and the struct format we use to unmarshal, we can make them match, like it's done in the `setLimits` function....

Reply:
No we can't because of the customResources field, that can be anything, like gpu's or something custom, so we can't have that because the struct would need to be dynamic

Reply:
requests.nvidia.com/gpu I guess is a standard one, although not very common so I didn't include it but you can have resource quotas on completely custom resources such as count/widgets.example.com or wheels/custom.com

Reply:
Sure, but all the other parts can be marshaled and this appended at the end like you are doing now.

Reply:
so two questions

1, how would I tell it to marshal everything but the customResources?

2, this is what the actual yaml looks like for k8s
status:
  hard:
    limits.cpu: "10"
    limits.memory: 8Gi
    pods: "25"
    requests.cpu: "10"
    requests.memory: 8Gi 

So I would have to change the structure of the helmsman yaml would have to change, we can do that but I think it'd be a little weird to have the dot notation

Reply:
@luisdavim hey this should be resolved now, sorry my day job was very busy this last month. Let me know if the way I handled not marshaling the custom quotas is okay